test/WorkflowCore.UnitTests/MemoryPersistenceProviderTests/MemoryPersistenceProviderTests.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Threading.Tasks;
+using FluentAssertions;
+using WorkflowCore.Exceptions;
+using WorkflowCore.Models;
+using WorkflowCore.Services;
+using Xunit;
+
+namespace WorkflowCore.UnitTests.MemoryPersistenceProviderTests;
+
+public class MemoryPersistenceProviderTests
+{
+    private readonly MemoryPersistenceProvider _provider = new();
+
+    [Fact]
+    public void Should_Throw_If_Workflow_With_Same_Reference_Already_Exist()
```

Comment:
It's already covered by the existing test, so I'll delete this file from PR.
![image](https://github.com/SergiiKram/workflow-core/assets/2759478/d2630fe9-2839-4f27-b792-3fff4027f713)